dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/util/FeatureInfo.java

```diff
@@ -0,0 +1,601 @@
+            }
+
+            String otherType = other.type;
+            if (type == null) {
+                if (otherType != null) {
+                    return false;
+                }
+            } else if (otherType == null) {
+                return false;
+            } else if (!type.equals(otherType)) {
+                return false;
+            }
+
+            return true;
+        }
+
+        private final String asString;
+
+        @Override
+        public String toString() {
+            return asString;
+        }
+    }
+
+    private static List<ExternalPackageInfo> parseExternalPackages(String packageList, String defaultType) {
+        if (packageList == null) {
+            return null;
+        }
+
+        List<ExternalPackageInfo> storage = new ArrayList<>();
+
+        String[] packageNames = packageList.split(",");
+        for (String packageName : packageNames) {
+            String[] packageParts = packageName.split(";");
+
+            String externalPackage = packageParts[0].trim();
+
+            String type = null;
+            if (packageParts.length > 1) {
+                for (int partNo = 1; partNo < packageParts.length; ++partNo) {
+                    String packagePart = packageParts[partNo].trim();
+                    if (packagePart.startsWith("type")) {
+                        type = packagePart.substring(packagePart.indexOf('=') + 1).trim();
+                        while (type.startsWith("\"")) {
+                            type = type.substring(1);
+                        }
+                        while (type.endsWith("\"")) {
+                            type = type.substring(0, type.length() - 1);
+                        }
+                        break;
+                    }
+                }
+            }
+
+            if (type == null) {
+                type = defaultType;
+            }
+
+            storage.add(new ExternalPackageInfo(externalPackage, type));
+        }
+
+        return new ArrayList<>(storage);
+    }
+
+    //
+
+    public static List<String> parseList(String raw) {
+        if (raw == null) {
+            return null;
+        }
+
+        String[] rawElements = raw.split(",");
+        List<String> asList = new ArrayList<>(rawElements.length);
+        for (String rawElement : rawElements) {
+            rawElement = rawElement.trim();
+            if (!rawElement.isEmpty()) {
+                asList.add(rawElement);
+            }
+        }
+        return (asList.isEmpty() ? null : asList);
+    }
+
+    public static String getBaseName(String featureName) {
+        int versionIndex = featureName.lastIndexOf('-');
+        if (versionIndex != -1) {
+            return featureName.substring(0, versionIndex);
+        } else {
+            return featureName;
+        }
+    }
+
+    public static String getVersion(String featureName) {
+        int versionIndex = featureName.lastIndexOf('-');
+        if (versionIndex != -1) {
```

Comment:
May want to import `org.osgi.framework.Version` and use `Version.parseVersion()`